cloudbuild.yaml

```diff
@@ -31,86 +31,28 @@ steps:
     timeout: 5m
     args: ['./scripts/git_init']
 
-  # Make sure that the Bazel server is unpacked and initialized, by calling `bazel info` (any other
-  # Bazel command would probably do).
   - name: 'gcr.io/oak-ci/oak:latest'
-    id: bazel_init
-    entrypoint: 'bazel'
+    id: runner_ci
     waitFor: ['git_init']
-    timeout: 5m
-    args: ['info']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: cargo_crev
-    entrypoint: 'bash'
-    waitFor: ['git_init']
-    timeout: 5m
-    args: ['./scripts/run_cargo_crev']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: build_server_no_debug
-    waitFor: ['bazel_init']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/build_server', '-s', 'logless']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: build_server
-    waitFor: ['build_server_no_debug']
     timeout: 60m
     entrypoint: 'bash'
-    args: ['./scripts/build_server', '-s', 'base']
+    args: ['./scripts/runner', '--commands', 'run-ci']
 
   # Package the Hello World application in a Docker image.
   - name: 'gcr.io/oak-ci/oak:latest'
     id: build_server_docker
-    waitFor: ['build_server']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/build_example', '-e', 'hello_world', '-i', 'base']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: run_tests
-    waitFor: ['build_server_docker']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/run_tests']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: run_tests_tsan
-    waitFor: ['run_tests']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/run_tests_tsan']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: run_examples
-    waitFor: ['run_tests_tsan']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/run_examples', '-s', 'base']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: run_examples_cpp
-    waitFor: ['run_examples']
-    timeout: 60m
-    entrypoint: 'bash'
-    args: ['./scripts/run_examples', '-s', 'base', '-a', 'cpp']
-
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: build_experimental
-    waitFor: ['run_tests']
+    waitFor: ['git_init']
     timeout: 60m
     entrypoint: 'bash'
-    args: ['./scripts/build_experimental']
-
-  # Run clang-tidy after the examples finish running, since they share the same `output_base`.
-  - name: 'gcr.io/oak-ci/oak:latest'
-    id: run_clang_tidy
-    waitFor: ['run_examples_cpp']
-    timeout: 30m
-    entrypoint: 'bash'
-    args: ['./scripts/run_clang_tidy']
+    args:
+      [
+        './scripts/runner',
+        'run-examples',
```

Comment:
Yeah there is definitely room for improvement. I was thinking to rename the subcommand to just `examples` and then have a flag `--run=[client|server|both|none]` (default `both`), or something along those lines? I think it would make the step-by-step docs more readable too. WDYT?